src/peekableView.ios.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import {
+  requireNativeComponent,
+  View,
+  StyleSheet,
+  findNodeHandle,
+} from 'react-native';
+
+export const ScreenContext = React.createContext();
+
+export const NativePeekableView = requireNativeComponent(
+  'RNSPeekableView',
+  null
+);
+
+export default class PeekableView extends React.Component {
+  static traverseActions(actions, actionsMap) {
+    const traversedAction = [];
+    if (!actions) {
+      return;
+    }
+    actions.forEach(currentAction => {
+      if (currentAction.group) {
+        const { group, ...clonedAction } = currentAction;
+        clonedAction['group'] = this.traverseActions(group, actionsMap);
+        traversedAction.push(clonedAction);
+      } else {
+        const { action, ...clonedAction } = currentAction;
+        clonedAction['_key'] = actionsMap.length;
+        actionsMap.push(action);
+        traversedAction.push(clonedAction);
+      }
+    });
+    return traversedAction;
+  }
+
+  static getDerivedStateFromProps(props) {
+    const mappedActions = [];
+    const traversedActions = PeekableView.traverseActions(
+      props.previewActions,
+      mappedActions
+    );
+    return {
+      traversedActions,
+      mappedActions,
+    };
+  }
+
+  preview = React.createRef();
+  previewView = React.createRef();
```

Comment:
this is unused. I am confused now because it seem like children are what you seem to treat like preview. So maybe it'd make more sense not to call it `child` but `preview` which would also carry some additional semantics as opposed to "children" which does not say much about what is it used for in this case